kodilla-stream/src/test/java/com/kodilla/stream/portfolio/BoardTestSuite.java

```diff
@@ -0,0 +1,171 @@
+                user4,
+                user2,
+                LocalDate.now().minusDays(15),
+                LocalDate.now().minusDays(2));
+
+        //taskLists
+        TaskList taskListToDo = new TaskList("To do");
+        taskListToDo.addTask(task1);
+        taskListToDo.addTask(task3);
+        TaskList taskListInProgress = new TaskList("In progress");
+        taskListInProgress.addTask(task5);
+        taskListInProgress.addTask(task4);
+        taskListInProgress.addTask(task2);
+        TaskList taskListDone = new TaskList("Done");
+        taskListDone.addTask(task6);
+
+        //board
+        Board project = new Board("Project Weather Prediction");
+        project.addTaskList(taskListToDo);
+        project.addTaskList(taskListInProgress);
+        project.addTaskList(taskListDone);
+        return project;
+    }
+
+    @Test
+    void testAddTaskListFindUsersTasks() {
+        //Given
+        Board project = prepareTestData();
+
+        //When
+        User user = new User("developer1", "John Smith");
+        List<Task> tasks =  project.getTaskLists().stream()
+                .flatMap(l -> l.getTasks().stream())
+                .filter(t ->t.getAssignedUser().equals(user))
+                .collect(toList());
+
+        //Then
+        assertEquals(2, tasks.size());
+        assertEquals(user, tasks.get(0).getAssignedUser());
+        assertEquals(user, tasks.get(1).getAssignedUser());
+    }
+
+    @Test
+    void testAddTaskListFindOutdatedTasks() {
+        //Given
+        Board project = prepareTestData();
+
+        //When
+        List<TaskList> undoneTasks = new ArrayList<>();
+        undoneTasks.add(new TaskList("To do"));
+        undoneTasks.add(new TaskList("In progress"));
+        List<Task> tasks = project.getTaskLists().stream()
+                .filter(undoneTasks::contains)
+                .flatMap(tl -> tl.getTasks().stream())
+                .filter(t -> t.getDeadline().isBefore(LocalDate.now()))
+                .collect(toList());
+
+        //Then
+        assertEquals(1, tasks.size());
+        assertEquals("HQLs for analysis", tasks.get(0).getTitle());
+    }
+
+    @Test
+    void testAddTaskListFindLongTasks() {
+        //Given
+        Board project = prepareTestData();
+
+        //When
+        List<TaskList> inProgressTasks = new ArrayList<>();
+        inProgressTasks.add(new TaskList("In progress"));
+        long longTasks = project.getTaskLists().stream()
+                .filter(inProgressTasks::contains)
+                .flatMap(tl -> tl.getTasks().stream())
+                .map(Task::getCreated)
+                .filter(d -> d.compareTo(LocalDate.now().minusDays(10)) <= 0)
+                .count();
+
+        //Then
+        assertEquals(2, longTasks);
+    }
+
+    @Test
+    void testAddTaskListAverageWorkingOnTask() {
+        //Given
+        Board project = prepareTestData();
+
+        //When
+        List<TaskList> inProgressTasks = new ArrayList<>();
```

Comment:
krócej:
List<TaskList> inProgressTasks = Arrays.asList(new TaskList("In progress"));